coreneuron/io/reports/nrnreport.hpp

```diff
@@ -51,7 +51,9 @@ namespace coreneuron {
 /// name of the variable in mod file used for setting synapse id
 #define SYNAPSE_ID_MOD_NAME "synapseID"
 
-enum ReportType { SomaReport, CompartmentReport, SynapseReport, IMembraneReport };
+enum ReportType { SomaReport, CompartmentReport, SynapseReport, IMembraneReport,
+                  AxonReport, DendriteReport, ApicalReport, AxonCompReport,
+                  DendriteCompReport, ApicalCompReport };
```

Comment:
nitpick : as we have more report types, here we should add some docstrings explaining what is different between DendriteReport vs DendriteCompReport etc.

Reply:
Is DendriteCompReport means all compartments of dendrite and DendriteReport means central compartment of dendrite?
In that case either we can rename it or define it with docs:

```
enum ReportType { 
    SomaReport,           // here my explanation
    CompartmentReport,    // another type explanation

```

Reply:
We have modified the `enum` to contain a `SectionReport` type, instead of mixing types and subtypes. This way, the new `SectionType` and the compartment property can define the purpose of the report.